### PR TITLE
Preference to convert single quote attribute values to double quote on format.

### LINKED ISSUE #263: Formatter surrounds single-quoted attributes with double-quotes
Formatting this  
```xml
<?xml version='1.0' encoding='UTF-8'?>
<?compositeMetadataRepository version='1.0.0'?>
<repository name='m2e-apt updates' type='org.eclipse.equinox.internal.p2.metadata.repository.CompositeMetadataRepository' version='1.0.0'>
<properties size='2'>
<property name='p2.compressed' value='true'/>
<property name='p2.timestamp' value='1328709112213'/>
</properties>
<children size='9'>
<child location='http://download.jboss.org/jbosstools/updates/m2e-extensions/m2e-apt/1.5.1-2018-11-16_19-54-26-H16/' />
<child location='http://download.jboss.org/jbosstools/updates/m2e-extensions/m2e-apt/1.5.0-2018-08-14_20-24-39-H12/' />
<child location='http://download.jboss.org/jbosstools/updates/m2e-extensions/m2e-apt/1.4.0-2018-01-12_16-25-02-H6/' />
<child location='http://download.jboss.org/jbosstools/updates/m2e-extensions/m2e-apt/1.3.0-2016-10-26_18-05-00-B50/' />
<child location='http://download.jboss.org/jbosstools/updates/m2e-extensions/m2e-apt/1.2.0-2016-01-25_16-03-44-H45/' />
<child location='http://download.jboss.org/jbosstools/updates/m2e-extensions/m2e-apt/1.1.1-2014-11-26_15-11-10-H37/' />
<child location='http://download.jboss.org/jbosstools/updates/m2e-extensions/m2e-apt/1.1.0-2014-05-21_05-04-36-H34/' />
<child location='http://download.jboss.org/jbosstools/updates/m2e-extensions/m2e-apt/1.0.1-2012-09-20_03-20-41-H23/'/>
<child location='http://download.jboss.org/jbosstools/updates/m2e-extensions/m2e-apt/1.0.0-2012-06-05_16-15-59-H16/'/>
</children>
</repository>
```

yields:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<?compositeMetadataRepository version='1.0.0' ?>
<repository name="'m2e-apt updates'" type="'org.eclipse.equinox.internal.p2.metadata.repository.CompositeMetadataRepository'" version="'1.0.0'">
    <properties size="'2'">
        <property name="'p2.compressed'" value="'true'" />
        <property name="'p2.timestamp'" value="'1328709112213'" />
    </properties>
    <children size="'9'">
        <child location="'http://download.jboss.org/jbosstools/updates/m2e-extensions/m2e-apt/1.5.1-2018-11-16_19-54-26-H16/'" />
        <child location="'http://download.jboss.org/jbosstools/updates/m2e-extensions/m2e-apt/1.5.0-2018-08-14_20-24-39-H12/'" />
        <child location="'http://download.jboss.org/jbosstools/updates/m2e-extensions/m2e-apt/1.4.0-2018-01-12_16-25-02-H6/'" />
        <child location="'http://download.jboss.org/jbosstools/updates/m2e-extensions/m2e-apt/1.3.0-2016-10-26_18-05-00-B50/'" />
        <child location="'http://download.jboss.org/jbosstools/updates/m2e-extensions/m2e-apt/1.2.0-2016-01-25_16-03-44-H45/'" />
        <child location="'http://download.jboss.org/jbosstools/updates/m2e-extensions/m2e-apt/1.1.1-2014-11-26_15-11-10-H37/'" />
        <child location="'http://download.jboss.org/jbosstools/updates/m2e-extensions/m2e-apt/1.1.0-2014-05-21_05-04-36-H34/'" />
        <child location="'http://download.jboss.org/jbosstools/updates/m2e-extensions/m2e-apt/1.0.1-2012-09-20_03-20-41-H23/'" />
        <child location="'http://download.jboss.org/jbosstools/updates/m2e-extensions/m2e-apt/1.0.0-2012-06-05_16-15-59-H16/'" />
    </children>
</repository>
```
single quotes should be either  ignored/reused or replaced

### PR DESCRIPTION
**Pull request by NikolasKomonen**
Fixes #263, #294

### CODE REVIEW

File: org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/settings/XMLFormattingOptions.java [hunk 1]
```diff
@@ -168,6 +177,46 @@ public boolean isSpaceBeforeEmptyCloseTag() {
...
+
+	/**
+	 * If the quotations preference is a valid option.
+	 * 
+	 * Keep up to date with new preferences.
+	 * @return
+	 */
+	private boolean isValidQuotations() {
+		final String value = this.getString(XMLFormattingOptions.QUOTATIONS);
+		return value != null && (value.equals(SINGLE_QUOTES_VALUE) || value.equals(DOUBLE_QUOTES_VALUE));
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/settings/XMLFormattingOptions.java]: return SINGLE_QUOTES_VALUE.equals(value) || DOUBLE_QUOTES_VALUE.equals(value);

File: org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/settings/XMLFormattingOptions.java [hunk 2]
```diff
@@ -168,6 +177,46 @@ public boolean isSpaceBeforeEmptyCloseTag() {
...
+	}
+
+	/**
+	 * Checks if {@code quotation} equals the current value for {@code format.quotations}.
+	 * @param quotation 
+	 * @return
+	 */
+	public boolean isQuotations(String quotation) {
+		String value = getQuotations();
+		return value.equals(quotation);
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/settings/XMLFormattingOptions.java]: `Objects.equals(value, quotation)` is safer

**Comment by NikolasKomonen** [comment]
@fbricon updated

File: org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/settings/SettingsTest.java [hunk 3]
```diff
@@ -103,19 +104,28 @@ public void formatSettings() {
...
+		
+		xmlFormattingOptions.setQuotations("InvalidValue"); // set a value that is not recognized
+
 		Assert.assertEquals(5, xmlFormattingOptions.getTabSize()); // value coming from the request formattingOptions
 		Assert.assertFalse(xmlFormattingOptions.isInsertSpaces()); // formattingOptions doesn't defines insert spaces
 																	// flag
 
 		Assert.assertFalse(xmlFormattingOptions.isJoinCommentLines());//Since default for JoinCommentLines is False
 
+		// Assumues the "InvalidValue" will be overridden
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/settings/SettingsTest.java]: Assumes

**Comment by fbricon** [comment]
we're trying to be too clever there:
feb-21-2019 17-59-12

**Comment by NikolasKomonen** [comment]
@fbricon updated

File: org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/dom/parser/XMLScanner.java [hunk 4]
```diff
@@ -315,25 +311,27 @@ TokenType internalScan() {
...
-				state = ScannerState.WithinTag;
-				hasSpaceAfterTag = false;
 				return finishToken(offset, TokenType.AttributeValue);
 			}
 			state = ScannerState.WithinTag;
-			hasSpaceAfterTag = false;
-			return internalScan(); // no advance yet - jump to WithinTag
+			return internalScan();
+
+			// int ch = stream.peekChar();
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/dom/parser/XMLScanner.java]: delete commented code

**Pull request merged**